python/examples/advanced/Segments.ipynb

```diff
@@ -0,0 +1,148 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Segments"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/whylabs/whylogs/blob/mainline/python/examples/advanced/Segments.ipynb)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "ANALYTICS_OPT_OUT = \"WHYLOGS_NO_ANALYTICS\"\n",
+    "os.environ['ANALYTICS_OPT_OUT']= \"1\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# %pip install whylogs[whylabs]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "\n",
+    "import whylogs as why\n",
+    "from whylogs.api.logger.result_set import SegmentedResultSet\n",
+    "from whylogs.core.metrics.metrics import CardinalityMetric, DistributionMetric\n",
+    "from whylogs.core.schema import DatasetSchema\n",
+    "from whylogs.core.segment import Segment\n",
+    "from whylogs.core.segmentation_partition import (\n",
+    "    ColumnMapperFunction,\n",
+    "    SegmentationPartition,\n",
+    "    SegmentFilter,\n",
+    "    segment_on_column,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "             col1          col2\n",
+      "count  100.000000    100.000000\n",
+      "mean    49.500000   3611.850000\n",
+      "std     29.011492   3264.992284\n",
+      "min      0.000000      0.000000\n",
+      "25%     24.750000    674.025000\n",
+      "50%     49.500000   2695.550000\n",
+      "75%     74.250000   6064.575000\n",
+      "max     99.000000  10781.100000\n",
+      "After profiling the result set has: 5 segments\n"
+     ]
+    }
+   ],
+   "source": [
+    "input_rows = 100\n",
+    "segment_column = \"col3\"\n",
+    "number_of_segments = 5\n",
+    "d = {\n",
+    "    \"col1\": [i for i in range(input_rows)],\n",
+    "    \"col2\": [i * i * 1.1 for i in range(input_rows)],\n",
+    "    segment_column: [f\"x{str(i%number_of_segments)}\" for i in range(input_rows)],\n",
+    "}\n",
+    "\n",
+    "df = pd.DataFrame(data=d)\n",
+    "print(df.describe())\n",
+    "segmentation_partition = segment_on_column(\"col3\")\n",
+    "segmentation_partition.filter = SegmentFilter(filter_function=lambda df: df.col1 > 49)\n",
```

Comment:
segmentation_partition is a dict, so this line breaks. should access the ['col3'] key, perhaps?

Reply:
Yeah good catch I have local changes that include uploading segments to whylabs, let me clean that up and update this example as the refactor updates are in the PR here.